src/Evolution/Imex/Protocols/ImplicitSector.hpp

```diff
@@ -0,0 +1,155 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <type_traits>
+
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TypeTraits/IsA.hpp"
+
+/// \cond
+class DataVector;
+template <typename X, typename Symm, typename IndexList>
+class Tensor;
+namespace Tags {
+template <typename Tag>
+struct Source;
+}  // namespace Tags
+/// \endcond
+
+namespace imex::protocols {
+/// Protocol for an implicit sector of an IMEX system.
+///
+/// An implicit sector describes the sources for one implicit solve
+/// performed during IMEX evolution.  A system may have multiple
+/// implicit sectors, but they must be independent, i.e., their
+/// sources must not depend on any tensors in other sectors.
+///
+/// Classes implementing this protocol must define:
+///
+/// * a `tensors` type alias of tags for the variables to be solved for
+///
+/// * an `initial_guess` type to be passed to `db::mutate_apply`,
+///   taking additional arguments for the inhomogeneous terms \f$X\f$
+///   and implicit weight \f$w\f$ in the equation to be solved:
+///   \f$u = X + w S(u)\f$.  (See example below.)  It must return a
+///   `std::vector<GuessResult>` indicating, for each point, whether
+///   the implicit equation has been solved analytically or whether
+///   the numerical solve should continue.  An empty return is
+///   equivalent to `GuessResult::InitialGuess` for every point.  When
+///   this is called, the sector variables will have the value from
+///   the explicit part of the time step.  This mutator will not be
+///   called if the implicit weight is zero, as a system-independent
```

Comment:
I think the comment after ".. weight is zero" is a little confusing, since implicit weight being zero means no implicit step is required, not that it's analytically invertible.